forc-util/src/bytecode.rs

```diff
@@ -0,0 +1,143 @@
+use anyhow::anyhow;
+use sha2::{Digest, Sha256};
+use std::fs::{self, File};
+use std::io::Read;
+use std::path::Path;
+
+/// Parses a bytecode file into an iterator of instructions and their corresponding bytes.
+pub fn parse_bytecode_to_instructions<P>(
+    path: P,
+) -> anyhow::Result<
+    impl Iterator<
+        Item = (
+            Result<fuel_asm::Instruction, fuel_asm::InvalidOpcode>,
+            &'static [u8],
+        ),
+    >,
+>
+where
+    P: AsRef<Path> + Clone,
+{
+    let mut f = File::open(path.clone())
+        .map_err(|_| anyhow!("{}: file not found", path.as_ref().to_string_lossy()))?;
+    let metadata = fs::metadata(path.clone())
+        .map_err(|_| anyhow!("{}: file not found", path.as_ref().to_string_lossy()))?;
+    let mut buffer = vec![0; metadata.len() as usize];
+    f.read_exact(&mut buffer).expect("buffer overflow");
+
+    let instructions = {
+        let buffer = Box::leak(buffer.into_boxed_slice()); // Leak the buffer to extend its lifetime
+        fuel_asm::from_bytes(buffer.iter().cloned()).zip(buffer.chunks(fuel_asm::Instruction::SIZE))
+    };
+
+    Ok(instructions)
+}
+
+/// Gets the bytecode ID from a bytecode file. The bytecode ID is the hash of the bytecode after removing the
+/// condigurables section, if any.
+pub fn get_bytecode_id<P>(path: P) -> anyhow::Result<String>
+where
+    P: AsRef<Path> + Clone,
+{
+    let mut instructions = parse_bytecode_to_instructions(path.clone())?;
+
+    // Collect the first four instructions into a temporary vector
+    let mut first_six_instructions = Vec::with_capacity(6);
+    for _ in 0..6 {
+        if let Some(instruction) = instructions.next() {
+            first_six_instructions.push(instruction);
+        } else {
+            return Err(anyhow!("Incomplete bytecode"));
+        }
+    }
```

Comment:
While perfectly fine, not sure if something like the following is tidier:

```
    let first_six_instructions = instructions.iter().take(6).copied().collect();
    ensure!(first_six_instructions.len() == 6, "Incomplete bytecode");
```

Reply:
`copied` doesn't work because of the static lifetime on the raw chunks. Plus, I think it's preferable to do it without copying the iterator.